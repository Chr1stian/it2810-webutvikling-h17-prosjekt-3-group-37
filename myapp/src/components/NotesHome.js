import React, { Component } from 'react';
import { CardText, CardTitle } from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';
import { Link } from 'react-router-dom';

class NotesHome extends Component {

  render(){
    let getValue = localStorage.getItem('value');
    let noteList = getValue || [];
    let noteText;
    if (noteList.length > 1) {
      noteText = (getValue.substring(0,50) + "...");
    }else{
      noteText = "You got no note."
    }
    //console.log(getValue.length);
    return(
      <div>
      <CardTitle style={{'paddingBottom':'0'}} title="Notes"></CardTitle>
      <CardText>{ noteText }</CardText>
      <RaisedButton style={{'display':'flex'}}>
        <Link className="navLink" to="/notes">EDIT YOUR NOTE</Link>
      </RaisedButton>
      </div>
    )


  }
}

export default NotesHome;
