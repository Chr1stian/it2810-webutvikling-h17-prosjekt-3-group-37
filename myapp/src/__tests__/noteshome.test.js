import React from "react";
//Import Enzyme with shallow to test component as a unit
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
//Importing component that will be tested
import NotesHome from "./../components/NotesHome.js";
import "./../test/mock-localstorage.js";

//takes snapshot of component
test("Component renders as expected", () => {
  const component = shallow(<NotesHome />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
