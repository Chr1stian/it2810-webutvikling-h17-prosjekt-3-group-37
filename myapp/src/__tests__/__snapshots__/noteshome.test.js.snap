// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div>
  <CardTitle
    style={
      Object {
        "paddingBottom": "4px",
        "paddingTop": "4px",
        "textAlign": "center",
      }
    }
    title="Notes"
  />
  <Divider
    inset={false}
  />
  <CardText
    className="noteHomeText"
  >
    You got no note.
  </CardText>
  <RaisedButton
    disabled={false}
    fullWidth={false}
    labelPosition="after"
    primary={false}
    secondary={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Link
      className="navLink"
      replace={false}
      to="/note"
    >
      EDIT YOUR NOTE
    </Link>
  </RaisedButton>
</div>
`;
